lib/roger/renderer.rb

```diff
@@ -158,6 +158,10 @@ def render_file(path, options = {})
       end
 
       template = template(pn.to_s, nil)
+
+      # Once I understand I will update this comment
+      template_nesting.push(template)
```

Comment:
`render_file` did not yet have support for nesting templates. This was an oversight so I think it is necessary to do this here.